### PR TITLE
Add the required dependencies to flight-desktop-restapi

### PR DESCRIPTION
**Pull request by WilliamMcCumstie**
`flight-desktop-restapi` requires websockify to connect to the VNC sessions. This is not available via  the base repos on `centos7` or `centos8`.

A new `flight-websockify` package has been created to provide this dependency. It has been built in broadly the same way as `flight-certbot` and uses `flight-python`.

A new release of `flight-desktop` is required to patch some hard coded paths. Ideally `flight-desktop` should be updated so it can be reconfigured.

---

The only other missing package is `xorg-x11-apps` on `centos8`. This package is available via `PowerTools`, but isn't installed in the base repo. For the time being, it is being redistributed via the `openflight` repos.

`xorg-x11-apps`/`x11-apps` are available via the base repos on `ubuntu` and `centos7` and have been added as dependencies.

---

`netpbm`/`netpbm-progs` is already available via the best repos on all distros, and has been added as a dependency.

### CODE REVIEW

File: builders/flight-websockify/config/projects/flight-websockify.rb [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+#
+#  https://opensource.org/licenses/EPL-2.0
+#
+# For more information on OpenFlight Omnibus Builder, please visit:
+# https://github.com/openflighthpc/openflight-omnibus-builder
+#===============================================================================
+name 'flight-websockify'
+maintainer 'Alces Flight Ltd'
+homepage 'https://github.com/openflighthpc/openflight-omnibus-builder/builders/flight-websockify'
+friendly_name 'Flight Certbot'
```
Comment (benarmston) [inline builders/flight-websockify/config/projects/flight-websockify.rb:30]: *Websockify

File: builders/flight-websockify/config/projects/flight-websockify.rb [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+VERSION = '1.0.0'
+PYTHON_SYSTEM = '3.8'
+override 'flight-websockify', version: VERSION
+override 'enforce-flight-python', version: PYTHON_SYSTEM
+
+build_version VERSION
+build_iteration 2
+
+override 'sqlite3', version: '3.32.3.0'
```
Comment (benarmston) [inline builders/flight-websockify/config/projects/flight-websockify.rb:42]: This can be removed.

File: builders/flight-websockify/lib/Pipfile [hunk 3]
```diff
@@ -0,0 +1,13 @@
+[[source]]
+name = "pypi"
+url = "https://pypi.org/simple"
+verify_ssl = true
+
+[dev-packages]
+
+[packages]
+numpy = "*"
+websockify = "*"
```
Comment (benarmston) [inline builders/flight-websockify/lib/Pipfile:10]: I'm not familiar with pipfiles.  Does this specify websockify as a dependency without specifying a version?

**Review by benarmston — COMMENTED** [review]
I'll push fixes for the two issues in a moment.  Can you take a look at the questions @WilliamMcCumstie ?

File: builders/flight-websockify/config/projects/flight-websockify.rb (hunk 1 above)
Comment (benarmston) [reply builders/flight-websockify/config/projects/flight-websockify.rb]: Addressed in 99b06e6f

File: builders/flight-websockify/config/projects/flight-websockify.rb (hunk 2 above)
Comment (benarmston) [reply builders/flight-websockify/config/projects/flight-websockify.rb]: Addressed in 99b06e6f

File: builders/flight-websockify/lib/Pipfile (hunk 3 above)
Comment (WilliamMcCumstie) [reply builders/flight-websockify/lib/Pipfile:10]: Yeah, much the same way as `Gemfile` can use `gem 'puma'` without a version. Ultimately the version is set in the `Pipfile.lock`.

**Comment by benarmston** [comment]
This is working well.  Need to fiddle with the branches to get this merged into `user-web-suite-2021.5`.  Need to merge the `flight-python` branch first.

**Pull request merged**